### PR TITLE
testing tic-tac-toe code with Rspec 

### PR DESCRIPTION
**Pull request by Byabasaija**
In this project, we created RSpec tests for the tic tac toe game. 
As required by the project we tested the public methods in tic tac toe game.
Any review comments about our work are highly appreciated.

### CODE REVIEW

File: spec/tictactoe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+describe Player do
+  let(:player) { Player.new('name', 'mark') }
+
+  it ' checks for player names' do
+    expect(player.name).to eq('name')
+  end
+
+  it ' checks for player mark' do
+    expect(player.mark).to eq('mark')
+  end
```
Comment (ioilmio) [inline spec/tictactoe_spec.rb]: Should check for X or O

**Review by ioilmio — CHANGES_REQUESTED** [review]
## Hello @ahmetbozaci and @Byabasaija 👋🏽 
## Nice job, your tests are working so far! 👏🏽 
## But you need to test all your methods, in your classes;
Namely,  you need to test in the game class:
- create table
- getnumber (use snake case for this change it to get_number)
- check_input
- change_player
- update_table
- draw_check
- play_again
## Remember to check also for negative cases

## Split your tests into two files one for the game and one for the player;
## Fix rubocop errors about require_relatives removing extensions at the end fo files name

## When you are ready you can submit for another review;

*For further info slack me @illuminatosalvatore*

**Comment by ahmetbozaci** [comment]
Dear TSE,

- We seperated game and player classes in different files.
- Checked "X" in player test.
- Fixed rubocop errors.

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hi @ioilmio  & @Byabasaija**

Great job with the project so far it looks amazing. you made most of the required changes, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] There are 8 methods in the game class, but you only tested for 3.
     Please make sure to test the others as well.

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Pull request closed without merging**